app/src/components/TestToast.vue

```diff
@@ -0,0 +1,39 @@
+<template>
+  <div v-if="visible">
+    <div :class="['alert', typeClass]">
+      <IconCheck v-if="type === ToastType.Success" />
+      <IconInfo v-if="type === ToastType.Info" />
+      <IconWarning v-if="type === ToastType.Warning" />
+      <IconError v-if="type === ToastType.Error" />
+      <span>{{ message }}</span>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { defineProps, ref, onMounted, computed } from 'vue'
+import { ToastType } from '@/types'
+import IconCheck from '@/components/icons/IconCheck.vue'
+import IconInfo from '@/components/icons/IconInfo.vue'
+import IconWarning from '@/components/icons/IconWarning.vue'
+import IconError from '@/components/icons/IconError.vue'
+
+const props = defineProps<{ type: ToastType; message: string; timeout: number }>()
```

Comment:
Here you should use an object with an interface. Instead of specifying everithing

Reply:
What, do you mean to define an interface for `defineProps` type?

Reply:
It's to reduce parameter coupling. 

Instead of this
```
function createUser(name, age, email, address) {
    // Function implementation
}
```
use this. 
So any update of the object don't update the declaration of the function
```
function createUser(user) {
    const { name, age, email, address } = user;
    // Function implementation
}
```

Reply:
Okay, I updated on the latest commit